### PR TITLE
Configure IStartupFilters in FIFO order #653

### PR DESCRIPTION
**Pull request by JunTaoLuo**
#653 

cc @Tratcher @davidfowl

### CODE REVIEW

**Comment by Tratcher** [comment]
@weidazhao

File: test/Microsoft.AspNetCore.TestHost.Tests/TestServerTests.cs [hunk 1]
```diff
@@ -275,7 +275,7 @@ public Action<IApplicationBuilder> Configure(Action<IApplicationBuilder> next)
             {
                 app.Run(context =>
                 {
-                    Assert.NotNull(context.RequestServices);
+                    Assert.Null(context.RequestServices);
```
Comment (muratg) [inline test/Microsoft.AspNetCore.TestHost.Tests/TestServerTests.cs]: Why did this change?

**Comment by davidfowl** [comment]
:shipit:

File: test/Microsoft.AspNetCore.TestHost.Tests/TestServerTests.cs (hunk 1 above)
Comment (JunTaoLuo) [inline test/Microsoft.AspNetCore.TestHost.Tests/TestServerTests.cs]: Because the test was wrong previously.

Comment (muratg) [inline test/Microsoft.AspNetCore.TestHost.Tests/TestServerTests.cs]: :+1: :shipit:

**Comment by weidazhao** [comment]
Thanks guys for the quick turnaround! I've verified on v1.0.0-rc2-20260. @davidfowl  @JunTaoLuo @muratg @Tratcher

**Pull request merged**